cirq-core/cirq/transformers/routing/route_circuit_cqc.py

```diff
@@ -0,0 +1,419 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Heuristic qubit routing algorithm based on arxiv:1902.08091."""
+
+from typing import Dict, List, Optional, Set, Tuple, TYPE_CHECKING
+from itertools import combinations
+import networkx as nx
+
+from cirq import circuits, ops, protocols
+from cirq.transformers import transformer_api, transformer_primitives
+from cirq.transformers.routing import mapping_manager, line_initial_mapper
+
+if TYPE_CHECKING:
+    import cirq
+
+QidPair = Tuple['cirq.Qid', 'cirq.Qid']
+
+
+def disjoint_pair_qubit_pair_combinations(qubit_pairs: List[QidPair]) -> List[Tuple[QidPair, ...]]:
```

Comment:
I think we can omit the "qubit_pair" from function name since the argument of the function clarifies that it expects qubit_pairs. The name right now is a mouthful. 

Maybe rename to the following:
```suggestion
def disjoint_nc2_combinations(qubit_pairs: List[QidPair]) -> List[Tuple[QidPair, QidPair]]:
```